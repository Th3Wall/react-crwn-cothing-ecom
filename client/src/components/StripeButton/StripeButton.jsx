import StripeCheckout from 'react-stripe-checkout';
import axios from 'axios';

const StripeButton = ({price}) => {
    const priceForStripe = price * 100;
    const publishedKey = 'pk_test_51IBr8VHqXDEPY8zA15Ee2YhUD4RZlfpEPYKi4A7sHMZInj5iRYswwMS8dn4o04Se2ophvJbUal1fAThQlWO8GQUz00ZN0Bd13Y';

    const onPaidCart = token => {
        axios({
            url: 'payment',
            method: 'post',
            data: {
                amount: priceForStripe,
                token
            }
        }).then(response => {
            alert('Payment Successful!')
        }).catch(error => {
            console.log('Payment error ', error);
            alert('There was an issue with your payment. Please make sure you use the provided credit card')
        })
    }

    return (
        <StripeCheckout
            label={'Pay Now'}
            name={'CRWN Clothing Ltd.'}
            description={`Your total is $${price}`}
            panelLabel={'Pay Now'}
            billingAddress
            shoppingAddress
            image={'https://svgshare.com/i/CUz.svg'}
            amount={priceForStripe}
            stripeKey={publishedKey}
            token={onPaidCart}
        />
    )
}

export default StripeButton
