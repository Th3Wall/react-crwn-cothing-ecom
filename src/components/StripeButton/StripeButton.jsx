import './stripeButton.scss';
import StripeCheckout from 'react-stripe-checkout';

const StripeButton = ({price}) => {
    const priceForStripe = price * 100;
    const publishedKey = 'pk_test_51IBr8VHqXDEPY8zA15Ee2YhUD4RZlfpEPYKi4A7sHMZInj5iRYswwMS8dn4o04Se2ophvJbUal1fAThQlWO8GQUz00ZN0Bd13Y';

    const onPaidCart = token => {
        console.log(token);
        alert('Payment successful!');
    }

    return (
        <StripeCheckout
            label={'Pay Now'}
            name={'CRWN Clothing Ltd.'}
            description={`Your total is $${price}`}
            panelLabel={'Pay Now'}
            billingAddress
            shoppingAddress
            image={'https://svgshare.com/i/CUz.svg'}
            amount={priceForStripe}
            stripeKey={publishedKey}
            token={onPaidCart}
        />
    )
}

export default StripeButton
