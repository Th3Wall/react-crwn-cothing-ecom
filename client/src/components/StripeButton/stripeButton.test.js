import React from 'react';
import { shallow } from 'enzyme';
import StripeButton from './StripeButton';

describe('StripeButton', () => {
    it('should display StripeButton component', () => {
        const mockPrice = 1245;
        const wrapper = shallow(<StripeButton {...mockPrice} />);
        expect(wrapper).toMatchSnapshot();
    })
})