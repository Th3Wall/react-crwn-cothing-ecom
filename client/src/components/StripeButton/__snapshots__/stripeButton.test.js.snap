// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StripeButton should display StripeButton component 1`] = `
<ReactStripeCheckout
  ComponentClass="span"
  amount={NaN}
  billingAddress={true}
  className="StripeCheckout"
  description="Your total is $undefined"
  image="https://svgshare.com/i/CUz.svg"
  label="Pay Now"
  locale="auto"
  name="CRWN Clothing Ltd."
  panelLabel="Pay Now"
  reconfigureOnUpdate={false}
  shoppingAddress={true}
  stripeKey="pk_test_51IBr8VHqXDEPY8zA15Ee2YhUD4RZlfpEPYKi4A7sHMZInj5iRYswwMS8dn4o04Se2ophvJbUal1fAThQlWO8GQUz00ZN0Bd13Y"
  token={[Function]}
  triggerEvent="onClick"
/>
`;
